pkg/types/report.go

```diff
@@ -53,6 +53,7 @@ type Result struct {
 	Target            string                     `json:"Target"`
 	Class             ResultClass                `json:"Class,omitempty"`
 	Type              string                     `json:"Type,omitempty"`
+	Layer             string                     `json:"Layer,omitempty"` // Used only for secrets, as Trivy keeps secrets from all layers.
```

Comment:
We should have `Layer` here rather than in `Result`.
https://github.com/aquasecurity/trivy/blob/7699153c6679dd6eaf377700b5f587cb1741368d/pkg/fanal/types/secret.go#L11-L20

If the finding is the same, the first introduced layer should be taken.

- Layer 1
    - no secret found
- Layer 2
    - secret.txt
        - GitHub PAT
- Layer 3
    - secret.txt
        - GitHub PAT
        - AWS Access Key
- Layer 4
    - remove secret.txt

The result would be as below.

```
{
  "Target": "secret.txt",
  "Secrets": [
    {
      "RuleID": "github-pat",
      ...
      "Layer": "Layer 2",
    },
    {
      "RuleID": "aws-access-key",
      ...
      "Layer": "Layer 3",
    }
  ]
}
```

We should not have `github-pat` in `Layer 3`.

Reply:
Note that `Layer` actually consists of `Digest` and `DiffID`. The above example is for sake of simplicity.

Reply:
Done